src/Core/AuthenticationHelpers/ClientRoleHeaderAuthenticationMiddleware.cs

```diff
@@ -151,6 +170,34 @@ public static bool IsSystemRole(string roleName)
         return roleName.Equals(AuthorizationType.Authenticated.ToString(), StringComparison.OrdinalIgnoreCase) ||
                 roleName.Equals(AuthorizationType.Anonymous.ToString(), StringComparison.OrdinalIgnoreCase);
     }
+
+    /// <summary>
+    /// Uses the dab-config.json's Authentication provider name to resolve the
+    /// authentication scheme to use with httpContext.AuthenticateAsync(scheme).
+    /// </summary>
+    /// <param name="configuredProviderName">Dab config defined authentication provider name.</param>
+    /// <returns>Authentication Scheme</returns>
+    private static string ResolveConfiguredAuthNScheme(string? configuredProviderName)
+    {
+        switch (configuredProviderName)
+        {
+            case AuthenticationOptions.SIMULATOR_AUTHENTICATION:
+                return SimulatorAuthenticationDefaults.AUTHENTICATIONSCHEME;
+            case nameof(EasyAuthType.AppService):
+                return EasyAuthAuthenticationDefaults.APPSERVICEAUTHSCHEME;
+            case nameof(EasyAuthType.StaticWebApps):
+                return EasyAuthAuthenticationDefaults.SWAAUTHSCHEME;
+            case "AzureAD":
+            case "EntraID":
```

Comment:
should we make these constants and use case-insensitive matching?

Reply:
yes, good call and this is absolutely necessary. added.